docs/serialization.md

```diff
@@ -15,7 +15,15 @@ The header is structured like the following table:
 | `1` | minor version number |
 | `1` | patch version number |
 | varint | the length of the encoding name |
-| | the encoding name |
+| string | the encoding name |
+| varint | number of errors |
+| varint | byte length of error |
+| string | error string, as byte[] in source encoding |
+| ... | more errors |
+| varint | number of warnings |
+| varint | byte length of warning |
+| string | warning string, as byte[] in source encoding |
+| ... | more warnings |
```

Comment:
I have been thinking a nicer way to document this would be to have some "meta types" for the 1st column that we would declare above this table.
So for example we could have `string` which is `varint size + bytes` and we could then describe errors as `varint number of errors + string* errors`.
And once we dump location too, we could have `diagnostic` which is `location location, string message` and location is `varint start, varint length`.